osu.Game/Overlays/Mods/ModSelectOverlay.cs

```diff
@@ -438,7 +445,8 @@ private void updateFromExternalSelection()
                 }
             }
 
-            SelectedMods.Value = newSelection;
+            if (!SelectedMods.Disabled)
+                SelectedMods.Value = newSelection;
```

Comment:
This guard causes the mod select overlay to potentially enter an inconsistent state wherein the mods that get activated do not actually match `SelectedMods`. I really don't like that this is possible with this change.

Reply:
I agree it's unfortunate. I think it's only required to test the new test with the old code, so it can likely be reverted.

Reply:
If it can be reverted I think I could be fine with the rest of the changes, but I did try it and I couldn't get the test to not fail.